config/database.yml

```diff
@@ -7,12 +7,14 @@ default: &default
 development:
   <<: *default
   database: publishing_api_development
-  url: <%= ENV["DATABASE_URL"]%>
+  url: <%= ENV["DATABASE_URL"] %>
 
 test:
   <<: *default
-  database: publishing_api_test
-  url: <%= ENV["DATABASE_URL"].try(:sub, /([-_]development)?$/, '_test')%>
+  # EXECUTOR_NUMBER is a env var added by Jenkins, this allows concurrent builds
+  # on the same box without the database colliding.
+  database: publishing_api_test<%= "_executor_#{ENV['EXECUTOR_NUMBER']}" if ENV["EXECUTOR_NUMBER"] %>
+  url: <%= ENV["DATABASE_URL"].try(:sub, /([-_]development)?$/, "_test") %>
```

Comment:
the url parameter here will override the values set in database, username, password (and probably a few others). If it is nil no overriding will occur, which is what will normally be the case when you're running these tests.

Whereas if you want to run these tests in the docker environment for instance you'll need a url set and will want to run the substitution.

tl;dr - the env var is normally nil